### PR TITLE
add ability to change MergeOptions for Patch Strategic Merge

### PR DESCRIPTION
**Pull request by awakair**
At the moment, we have the option to append/prepend elements in the Merge function (as seen in https://github.com/kubernetes-sigs/kustomize/pull/3048). However, Patch Strategic Merge currently lacks an alternative for prepending elements in a list. This pull request will add the option for Patch Strategic Merge to append elements while merging, and it will forward this option up to `krusty`.

### CODE REVIEW

File: api/filters/patchstrategicmerge/patchstrategicmerge_test.go [hunk 1]
```diff
@@ -921,3 +921,150 @@ data:
 		})
 	}
 }
+
+func TestFilterMergeOptionsListAppend(t *testing.T) {
```
Comment (chansuke) [inline api/filters/patchstrategicmerge/patchstrategicmerge_test.go:925]: Thank you so much for your PR.
i would suggest considering the addition of some edge case tests:

- append to empty list
- append duplicate element

etc. IMHO

File: api/krusty/baseandoverlaymedium_test.go [hunk 2]
```diff
@@ -305,3 +306,199 @@ metadata:
   name: test-infra-app-config-49d6f5h7b5
 `)
 }
+
+func TestMediumOverlayMergeOptionsListAppend(t *testing.T) {
```
Comment (chansuke) [inline api/krusty/baseandoverlaymedium_test.go:310]: same here?

**Comment by awakair** [comment]
Thank you for your response! I added edge cases to tests

**Review by chansuke — APPROVED** [review]
LGTM

**Comment by chansuke** [comment]
@awakair 
Thank you so much for your work. Looks good for me:)

**Comment by awakair** [comment]
@chansuke
Thank you for quick response! What else should I do to merge this PR? Should I get `lgtm` label or something?

**Comment by devkpv** [comment]
Hey @varshaprasad96, could you please take a look when you get a chance?
